File: 30DaysOfCode/day09_recursion/recursion.pl
#!/usr/bin/perl

sub factorial {
    my $num = shift;
    if ($num <= 1) {
        return 1;
    }
    
    return $num * factorial($num - 1);
}

$n = <STDIN>;
chomp $n;
$result = factorial($n);
print "$result\n";

File: 30DaysOfCode/day09_recursion/recursion.php
<?php

$handle = fopen ("php://stdin", "r");
function factorial($n) {
    if ($n <= 1) {
        return 1;
    }
    return $n * factorial($n - 1);
}

fscanf($handle, "%i",$n);
$result = factorial($n);
echo $result . "\n";

?>
